### PR TITLE
[jsk_pcl_ros/icp_registration] Fix error in case of input point cloud…

### PR DESCRIPTION
**Pull request by iory**
Fix error in case of input point cloud size is 0.
I will write test code.

### CODE REVIEW

**Comment by k-okada** [comment]
can we merge this?

File: jsk_pcl_ros/src/icp_registration_nodelet.cpp [hunk 1]
```diff
@@ -455,6 +455,13 @@ namespace jsk_pcl_ros
       }
     }
     
+    if (min_score == DBL_MAX) {
```
Comment (wkentaro) [inline jsk_pcl_ros/src/icp_registration_nodelet.cpp]: Why not `if (cloud.empty()) {...}` before the for loop above?

Comment (iory) [inline jsk_pcl_ros/src/icp_registration_nodelet.cpp]: cloud is reference_cloud_list_?

Comment (wkentaro) [inline jsk_pcl_ros/src/icp_registration_nodelet.cpp]: yea

Comment (wkentaro) [inline jsk_pcl_ros/src/icp_registration_nodelet.cpp]: btw what error do you have if you have empty cloud?

Comment (iory) [inline jsk_pcl_ros/src/icp_registration_nodelet.cpp]: [FATAL] [1472562909.537805091]: [jsk_pcl_ros::ICPRegistration::align] no reference is specified
[ERROR] [1472562909.588487018]: [jsk_pcl_ros::ICPRegistration::alignPointcloudWithICP] Input Target Cloud is empty!
[ERROR] [1472562909.588821610]: [jsk_pcl_ros::ICPRegistration::alignPointcloudWithICP] Input Target Cloud is empty!
[ INFO] [1472562909.588895190]: [jsk_pcl_ros::ICPRegistration::scorePointcloudAlignment] flipped score: 179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000
[ INFO] [1472562909.588938766]: [jsk_pcl_ros::ICPRegistration::alignPointcloudWithReferences] best score is: 179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000
nodelet: /usr/include/boost/smart_ptr/shared_ptr.hpp:646: typename boost::detail::sp_dereference<T>::type boost::shared_ptr<T>::operator*() const [with T = pcl::PointCloud<pcl::PointXYZRGBNormal>; typename boost::detail::sp_dereference<T>::type = pcl::PointCloud<pcl::PointXYZRGBNormal>&]: Assertion `px != 0' failed.

**Comment by furushchev** [comment]
+1 for original @iory PR. 
We have to handle all cases that returns `DBL_MAX` in icp process.

File: jsk_pcl_ros/src/icp_registration_nodelet.cpp (hunk 1 above)
Comment (iory) [inline jsk_pcl_ros/src/icp_registration_nodelet.cpp]: This error occurs only when cloud->empty() is true.
so modified code.

**Comment by iory** [comment]
Please check this PR.

**Comment by k-okada** [comment]
@furushchev @wkentaro is this ok for you?

**Comment by wkentaro** [comment]
lgtm

**Comment by knorth55** [comment]
@iory 
why does this sample launch try to register drill while there is no drill?
is there no rosbag file with drill pointcloud?
icp_sample

**Pull request merged**